pkg/rpc/server/listen.go

```diff
@@ -111,6 +126,9 @@ func (w dbWorker) update(ctx context.Context, appVersion, cacheDir string,
 	if err = w.hotUpdate(ctx, cacheDir, dbUpdateWg, requestWg); err != nil {
 		return xerrors.Errorf("failed DB hot update")
 	}
+	if err = updateLastDBUpdatePrometheus(gaugeMetric, float64(time.Now().Unix()), false); err != nil { //updating the last_db_update metric (here, coz code reaches till here, then it's a successful db update)
+		return xerrors.Errorf("%w", err)
```

Comment:
Would you wrap this error with some context?

Reply:
actually, it was an old piece of code which I forgot to refactor. I'll deal with it.